disnake/welcome_screen.py

```diff
@@ -0,0 +1,223 @@
+            "channel_id": self.id,
+            "description": self.description,
+        }
+        if self.emoji is not None:
+            if self.emoji.id:
+                result["emoji_id"] = self.emoji.id
+            result["emoji_name"] = self.emoji.name
+
+        return result  # type: ignore
+
+
+class WelcomeScreen:
+    """Represents a Discord welcome screen for a :class:`Guild`.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    description: Optional[:class:`str`]
+        The guild description in the welcome screen.
+    channels: List[:class:`WelcomeScreenChannel`]
+        The welcome screen's channels.
+    """
+
+    __slots__ = (
+        "description",
+        "welcome_channels",
+        "_guild",
+        "_state",
+    )
+
+    def __init__(
+        self,
+        *,
+        data: WelcomeScreenPayload,
+        state: ConnectionState,
+        guild: Union[Guild, PartialInviteGuild],
+    ):
+        self._state = state
+        self._guild = guild
+        self.description: Optional[str] = data.get("description")
+        self.welcome_channels: List[WelcomeScreenChannel] = [
+            WelcomeScreenChannel._from_data(data=channel, state=state)
+            for channel in data["welcome_channels"]
+        ]
+
+    def __repr__(self) -> str:
+        return f"<WelcomeScreen description={self.description!r} welcome_channels={self.welcome_channels!r} enabled={self.enabled!r}>"
+
+    @property
+    def enabled(self) -> bool:
+        """:class:`bool`: Whether the welcome screen is displayed to users.
+        This is equivalent to checking if ``WELCOME_SCREEN_ENABLED`` is present in :attr:`Guild.features`.
+        """
+        return "WELCOME_SCREEN_ENABLED" in self._guild.features
+
+    async def edit(
+        self,
+        *,
+        enabled: bool = MISSING,
+        description: str = MISSING,
+        welcome_channels: List[WelcomeScreenChannel] = MISSING,
+        reason: Optional[str] = None,
+    ) -> WelcomeScreen:
+        """|coro|
+
+        Edits the welcome screen.
+
+        You must have the :attr:`~Permissions.manage_guild` permission to
+        use this.
+
+        All fields are optional and nullable.
+
+        Parameters
+        ----------
+        enabled: :class:`bool`
+            Whether the welcome screen is enabled.
+        description: :class:`str`
+            The new guild description in the welcome screen.
+        welcome_channels: List[:class:`WelcomeScreenChannel`]
+            The new welcome channels.
+        reason: Optional[:class:`str`]
+            The reason for editing the welcome screen. Shows up on the audit log.
+
+        Raises
+        ------
+        Forbidden
+            You do not have permissions to change the welcome screen.
+        HTTPException
+            Editing the welcome screen failed.
+
+        Returns
+        -------
+        :class:`WelcomeScreen`
+            The newly edited welcome screen.
+        """
+        payload = {}
```

Comment:
Any reason to not just call `Guild.edit_welcome_screen` here?

Reply:
https://github.com/DisnakeDev/disnake/pull/339/commits/3a3575c869d22c90a89e97947811ef198448948f